packages/statemanager/src/accessWitness.ts

```diff
@@ -0,0 +1,433 @@
+import { BIGINT_0, bytesToBigInt, bytesToHex, hexToBytes, toBytes } from '@ethereumjs/util'
+import { getKey, getStem } from '@ethereumjs/verkle'
+import debugDefault from 'debug'
+
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+const { debug: createDebugLogger } = debugDefault
+const debug = createDebugLogger('statemanager:verkle:aw')
+
+/**
+ * Tree key constants.
+ */
+export const VERSION_LEAF_KEY = toBytes(0)
+export const BALANCE_LEAF_KEY = toBytes(1)
+export const NONCE_LEAF_KEY = toBytes(2)
+export const CODE_KECCAK_LEAF_KEY = toBytes(3)
+export const CODE_SIZE_LEAF_KEY = toBytes(4)
+
+export const HEADER_STORAGE_OFFSET = 64
+export const CODE_OFFSET = 128
+export const VERKLE_NODE_WIDTH = 256
+// export const MAIN_STORAGE_OFFSET = BigInt(256) ** BigInt(31)
+// incorrect value to match with kaustinen2 offset
+export const MAIN_STORAGE_OFFSET = BigInt(256) * BigInt(2) ** BigInt(31)
+
+const WitnessBranchReadCost = BigInt(1900)
+const WitnessChunkReadCost = BigInt(200)
+const WitnessBranchWriteCost = BigInt(3000)
+const WitnessChunkWriteCost = BigInt(500)
+const WitnessChunkFillCost = BigInt(6200)
+
+// read is a default access event if stem or chunk is present
+type StemAccessEvent = { write?: boolean }
+// chunk fill access event is not being charged right now in kaustinen2 but will be rectified
+// in upcoming iterations
+type ChunkAccessEvent = StemAccessEvent & { fill?: boolean }
+
+type AccessEventFlags = {
+  stemRead: boolean
+  stemWrite: boolean
+  chunkRead: boolean
+  chunkWrite: boolean
+  chunkFill: boolean
+}
+
+// Since stem is pedersen hashed, it is useful to maintain the reverse relationship
+type StemMeta = { address: Address; treeIndex: number | bigint }
+type RawAccessedState = {
+  address: Address
+  treeIndex: number | bigint
+  chunkIndex: number
+  chunkKey: PrefixedHexString
+}
+
+export enum AccessedStateType {
```

Comment:
Can all exported members of this file please get a code code that these code parts are part of verkle implementation and that this is experimental and code/API can change (or something like that)?

Reply:
(maybe a generic two-liner like

> Experimental. This code is part of the Verkle [EIP-23423](link) implementation and the exposed API will change over time. Do not use in production.

or something like that to be repeated everywhere we need this, than this can be easily removed again by search-and-replace later on)